### PR TITLE
move ERB from the YAML header into the template body

### PR DESCRIPTION
**Pull request by Hamms**
# Description

As preparation for the work to ultimately remove ERB functionality from the YAML header entirely

## Links

- [jira](https://codedotorg.atlassian.net/browse/FND-798)

## Reviewer Checklist

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by joshlory** [comment]
I like this approach! Does this cover all non-markdown files with dynamic YAML headers?

**Comment by Hamms** [comment]
Not yet all; I'm still tracking down and fixing a few stragglers

File: pegasus/sites.v3/code.org/public/athletes.haml [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
+  @header["social"]["og:title"] = I18n.t(:sports_og_title)
+  @header["social"]["og:description"] = I18n.t(:sports_og_description)
+  @header["social"]["og:image"] = 'https://' + request.host + '/images/social-media/athletes-thumbnail.jpg'
+  @header["social"]["og:video:url"] = 'http://youtube.com/v/ip051U7Rvds'
+  @header["social"]["og:video:secure_url"] = 'https://youtube.com/v/ip051U7Rvds'
+  @header["social"]["twitter:title"] = I18n.t(:sports_og_title)
+  @header["social"]["twitter:description"] = I18n.t(:sports_og_description)
+  @header["social"]["twitter:image:src"] = 'https://' + request.host + '/images/social-media/athletes-thumbnail.jpg'
+  @header["social"]["twitter:player"] = 'https://youtube.com/embed/ip051U7Rvds'
+
```
Comment (breville) [inline pegasus/sites.v3/code.org/public/athletes.haml:27]: This page might be a good candidate to go into `social_metadata.rb`.

Comment (Hamms) [reply pegasus/sites.v3/code.org/public/athletes.haml:27]: I'm inclined to do that as follow-up work; keep this PR focused on maintaining existing functionality

File: pegasus/sites.v3/code.org/public/hourofcode/overview.haml [hunk 2]
```diff
@@ -2,18 +2,20 @@
...
-  "twitter:url": "https://<%=request.host%>/hourofcode/overview"
-  "twitter:title": "<%= hoc_s(:hour_of_code) %>"
-  "twitter:description": "<%= hoc_s(:social_hoc2018_global_movement) %>"
-  "twitter:image:src": "<%= 'https://' + request.host + '/images/social-media/code-2018-creativity.png' %>"
-  "og:title": "<%= hoc_s(:hour_of_code) %>"
-  "og:description": "<%= hoc_s(:social_hoc2018_global_movement) %>"
-  "og:image" : "<%= 'https://' + request.host + '/images/social-media/code-2018-creativity.png' %>"
-  "og:image:width": '1200'
-  "og:image:height": '630'
+  twitter:card: photo
```
Comment (breville) [inline pegasus/sites.v3/code.org/public/hourofcode/overview.haml:5]: It feels slightly disjointed to leave some of this metadata behind.  Would it be more readable to just move it all to the `:ruby` block below?

**Review by breville — APPROVED** [review]
Nice work on such a big change!  A couple comments but neither a deal-breaker on making forward progress here.

File: pegasus/sites.v3/code.org/public/hourofcode/overview.haml (hunk 2 above)
Comment (Hamms) [reply pegasus/sites.v3/code.org/public/hourofcode/overview.haml:5]: Yeah, I hear ya. I'm definitely of two minds on this; it doesn't feel great in general that we have these two pretty different ways of setting metadata, but as long as we do have both I think we should be okay with supporting both simultaneously.

Unless you have objections, I think I'd like to leave this as-is for now, and think of the work to move everything into the `:ruby` block as being tied in with work to explore deprecation of the yaml header entirely.

Comment (Hamms) [reply pegasus/sites.v3/code.org/public/hourofcode/overview.haml:5]: Tracking [here](https://codedotorg.atlassian.net/browse/FND-809)

File: pegasus/sites.v3/code.org/public/athletes.haml (hunk 1 above)
Comment (Hamms) [reply pegasus/sites.v3/code.org/public/athletes.haml:27]: Tracking [here](https://codedotorg.atlassian.net/browse/FND-810)

**Pull request merged**